### ISSUE TITLE
[Android] Image gallery does not open on File Chooser click

### ISSUE THREAD

**Issue by fcatuhe** [lazaronixon/react-native-turbolinks#2]
Hello, first of all thank you very much for this project. We are looking at using it for a production app we own, where we have turbolinks-ios and turbolinks-android adapters already in place, but would like to have a common codebase, and react-native combined with turbolinks is the perfect solution!

At the moment, the image gallery does not open in Android when you click on an `input type="file"`. It works out of the box for iOS.

We already had this issue in our native turbolinks-android adapter, and this is the code we had to add in our `onCreate` activity:
```java
// Code to open library
webView.setWebChromeClient(new WebChromeClient() {
    //For Android 4.1+
    public void openFileChooser(ValueCallback<Uri> uploadMsg, String acceptType, String capture) {
        mUploadingFile = true;
        mUploadMessage = uploadMsg;
        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
        intent.addCategory(Intent.CATEGORY_OPENABLE);
        intent.setType("image/*");
        startActivityForResult(Intent.createChooser(intent, "File Browser"), FILECHOOSER_RESULTCODE);
    }

    //For Android 5.0+
    public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, FileChooserParams fileChooserParams) {
        mUploadingFile = true;
        if (uploadMessage != null) {
            uploadMessage.onReceiveValue(null);
            uploadMessage = null;
        }
        uploadMessage = filePathCallback;

        Intent intent = null;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
            intent = fileChooserParams.createIntent();
        }
        try {
            startActivityForResult(intent, REQUEST_SELECT_FILE);
        } catch (ActivityNotFoundException e) {
            uploadMessage = null;
            Toast.makeText(getApplicationContext(), "Cannot Open File Chooser", Toast.LENGTH_LONG)
                    .show();
            return false;
        }
        return true;
    }
});
```

I hope this code can help you implement this functionality for Android!

**Comment by lazaronixon** [comment]
Wow thank you.. it will available on next release

Sent from my iPhone

On 25 Jan 2018, at 06:18, François Catuhe <notifications@github.com<mailto:notifications@github.com>> wrote:


Hello, first of all thank you very much for this project. We are looking at using it for a production app we own, where we have turbolinks-ios and turbolinks-android adapters already in place, but would like to have a common codebase, and react-native combined with turbolinks is the perfect solution!

At the moment, the image gallery does not open in Android when you click on an input type="file". It works out of the box for iOS.

We already had this issue in our native turbolinks-android adapter, and this is the code we had to add in our onCreate activity:

// Code to open library
webView.setWebChromeClient(new WebChromeClient() {
    //For Android 4.1+
    public void openFileChooser(ValueCallback<Uri> uploadMsg, String acceptType, String capture) {
        mUploadingFile = true;
        mUploadMessage = uploadMsg;
        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
        intent.addCategory(Intent.CATEGORY_OPENABLE);
        intent.setType("image/*");
        startActivityForResult(Intent.createChooser(intent, "File Browser"), FILECHOOSER_RESULTCODE);
    }

    //For Android 5.0+
    public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, FileChooserParams fileChooserParams) {
        mUploadingFile = true;
        if (uploadMessage != null) {
            uploadMessage.onReceiveValue(null);
            uploadMessage = null;
        }
        uploadMessage = filePathCallback;

        Intent intent = null;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
            intent = fileChooserParams.createIntent();
        }
        try {
            startActivityForResult(intent, REQUEST_SELECT_FILE);
        } catch (ActivityNotFoundException e) {
            uploadMessage = null;
            Toast.makeText(getApplicationContext(), "Cannot Open File Chooser", Toast.LENGTH_LONG)
                    .show();
            return false;
        }
        return true;
    }
});

I hope this code can help you implement this functionality for Android!

**Comment by lazaronixon** [comment]
Fixed with  6f7a367cfd3718be0d28d36ec286abf93edd2151

**Comment by lazaronixon** [comment]
it seems a problem with turbolinks-android... I hope it will be fixed on V2... I recommend you use a native approach for that...
https://github.com/lazaronixon/react-native-turbolinks/wiki/Upload-file
https://github.com/turbolinks/turbolinks-android/issues/53
